tests/cases/conformance/esDecorators/classDeclaration/esDecorators-classDeclaration-exportModifier.2.ts

```diff
@@ -0,0 +1,31 @@
+// @target: esnext
+// @module: esnext
+// @filename: global.ts
+
+/** @type {*} */
+var dec;
+
+// @filename: file1.ts
+
+// ok
+@dec export class C1 { }
+
+// @filename: file2.ts
+
+// ok
+@dec export default class C2 {}
+
+// @filename: file3.ts
+
+// error
+export @dec default class C3 {}
+
+// @filename: file4.ts
+
+// ok
+export @dec class C4 {}
+
+// @filename: file5.ts
+
+// ok
+export default @dec class C5 {}

```

Comment:
What about `@dec export default @dec` and `@dec export @dec`?

Does this PR fix #52578?

Reply:
Yes, I'll add it.